testingt.go

```diff
@@ -0,0 +1,218 @@
+	Log(args ...interface{})
+	// Logf will log a formatted string to the current testing.T log if set, otherwise it will log
+	// to stdout
+	Logf(format string, args ...interface{})
+	// Error fails the current test and logs the provided arguments. Equivalent to calling Log then
+	// Fail.
+	Error(args ...interface{})
+	// Errorf fails the current test and logs the formatted message. Equivalent to calling Logf then
+	// Fail.
+	Errorf(format string, args ...interface{})
+	// Fail marks the current test as failed, but does not halt execution of the step.
+	Fail()
+	// FailNow marks the current test as failed and halts execution of the step.
+	FailNow()
+	// Fatal logs the provided arguments, marks the test as failed and halts execution of the step.
+	Fatal(args ...interface{})
+	// Fatal logs the formatted message, marks the test as failed and halts execution of the step.
+	Fatalf(format string, args ...interface{})
+	// Skip logs the provided arguments and marks the test as skipped but does not halt execution
+	// of the step.
+	Skip(args ...interface{})
+	// Skipf logs the formatted message and marks the test as skipped but does not halt execution
+	// of the step.
+	Skipf(format string, args ...interface{})
+	// SkipNow marks the current test as skipped and halts execution of the step.
+	SkipNow()
+	// Skipped returns true if the test has been marked as skipped.
+	Skipped() bool
+}
+
+// Logf will log test output. If called in the context of a test and testing.T has been registered,
+// this will log using the step's testing.T, else it will simply log to stdout.
+func Logf(ctx context.Context, format string, args ...interface{}) {
+	if t := getTestingT(ctx); t != nil {
+		t.Logf(format, args...)
+		return
+	}
+	fallbackLogf(format, args...)
+}
+
+// Log will log test output. If called in the context of a test and testing.T has been registered,
+// this will log using the step's testing.T, else it will simply log to stdout.
+func Log(ctx context.Context, args ...interface{}) {
+	if t := getTestingT(ctx); t != nil {
+		t.Log(args...)
+		return
+	}
+	fallbackLog(args...)
+}
+
+// LoggedMessages returns an array of any logged messages that have been recorded during the test
+// through calls to godog.Log / godog.Logf or via operations against godog.T(ctx)
+func LoggedMessages(ctx context.Context) []string {
+	if t := getTestingT(ctx); t != nil {
+		return t.logMessages
+	}
+	return nil
+}
+
+// errStopNow should be returned inside a panic within the test to immediately halt execution of that
+// test
+var errStopNow = fmt.Errorf("FailNow or SkipNow called")
+
+type testingT struct {
+	name         string
+	t            *testing.T
+	failed       bool
+	skipped      bool
+	failMessages []string
+	logMessages  []string
+}
+
+// check interface against our testingT and the upstream testing.B/F/T:
+var (
+	_ TestingT = &testingT{}
+	_ TestingT = (*testing.B)(nil)
+	_ TestingT = (*testing.F)(nil)
```

Comment:
```suggestion
```

CI currently requires tests to pass in go1.16 (which might need to be upgraded to newer versions, but that's a different topic), let's remove this line to fix CI.

Reply:
done - 939bd89